instagram/src/components/Authentication/Authenticate.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import Login from '../Login/Login';
+import './Authenticate.css';
+
+
+const Authenticate = PassedComponent =>
+  class extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {loggedIN: false}
+    }
+
+    componentDidMount() {
+      if (localStorage.getItem('user' === 'ImaHacker') &&
+         localStorage.getItem('password') === 'password'
+      ) {
+        this.setState({loggedIN: true})
+      }
+      else {this.setState({loggedIN: false})
+      }
+
+    }
+
+    render() {
+      if(this.state.loggedIN) return <PassedComponent/>;
```

Comment:
I suggest using a ternary statement here instead of an if statement. `this.state.loggedIN ? <PassedComponent /> : <Login />`